### ISSUE TITLE
Use from behind a proxy that requires an authentication ?

### ISSUE THREAD

**Issue by nsebban** [siznax/wptools#130]
Hello,

I read in the docs that I can configure wptools so it uses a proxy for HTTP request, but I can't seem to make it work with a proxy that requires an authentication. 

I have been using the "http://<user>:<pass>@host:port" syntax, but it doesn't take the user/pass into account. Is there a way to make it work ?

Thanks in advance.

**Comment by siznax** [comment]
Thanks for trying wptools, @nsebban!

I'll look into providing support for proxy auth with pycurl. We haven't tried that yet!

**Comment by siznax** [comment]
I've added the option to supply proxy auth parameters as a dict:

```python
import wptools
proxy = {'PROXY': 'http://example.com', 'PORT': 80, 'USERPWD': 'user:password'}
page = wptools.page('Main_Page')
page.get_query(proxy=proxy)
```

@nsebban please try these options and let us know if they work for you.

There are [more options we can try](https://curl.haxx.se/libcurl/c/curl_easy_setopt.html) as well.

**Comment by siznax** [comment]
This fix was released with v0.4.17.
